tobac/analysis/analysis.py

```diff
@@ -798,172 +563,6 @@ def nearestneighbordistance_histogram(
-        )
-        * 1000
-    )
-
-    dists2 = np.zeros(_2Dlat_coord.points.shape)
-    dists2[:, 0] = hdist2[:, 0]
-    dists2[:, -1] = hdist2[:, -1]
-    dists2[:, 1:-1] = (hdist2[:, 0:-1] + hdist2[:, 1:]) * 0.5
-
-    area = dists1 * dists2
-
-    return area
-
-
-def calculate_area(features, mask, method_area=None):
-    """Calculate the area of the segments for each feature.
-
-    Parameters
-    ----------
-    features : pandas.DataFrame
-        DataFrame of the features whose area is to be calculated.
-
-    mask : iris.cube.Cube
-        Cube containing mask (int for tracked volumes 0 everywhere
-        else). Needs to contain either projection_x_coordinate and
-        projection_y_coordinate or latitude and longitude
-        coordinates.
-
-    method_area : {None, 'xy', 'latlon'}, optional
-        Flag determining how the area is calculated. 'xy' uses the
-        areas of the individual pixels, 'latlon' uses the
-        area_weights method of iris.analysis.cartography, None
-        checks wether the required coordinates are present and
-        starts with 'xy'. Default is None.
-
-    Returns
-    -------
-    features : pandas.DataFrame
-        DataFrame of the features with a new column 'area',
-        containing the calculated areas.
-
-    Raises
-    ------
-    ValueError
-        If neither latitude/longitude nor
-        projection_x_coordinate/projection_y_coordinate are
-        present in mask_coords.
-
-        If latitude/longitude coordinates are 2D.
-
-        If latitude/longitude shapes are not supported.
-
-        If method is undefined, i.e. method is neither None,
-        'xy' nor 'latlon'.
-
-    """
-
-    from tobac.utils import mask_features_surface, mask_features
-    from iris import Constraint
-    from iris.analysis.cartography import area_weights
-    from scipy.ndimage import labeled_comprehension
-
-    features["area"] = np.nan
-
-    mask_coords = [coord.name() for coord in mask.coords()]
-    if method_area is None:
-        if ("projection_x_coordinate" in mask_coords) and (
-            "projection_y_coordinate" in mask_coords
-        ):
-            method_area = "xy"
-        elif ("latitude" in mask_coords) and ("longitude" in mask_coords):
-            method_area = "latlon"
-        else:
-            raise ValueError(
-                "either latitude/longitude or projection_x_coordinate/projection_y_coordinate have to be present to calculate distances"
-            )
-    # logging.debug("calculating area using method " + method_area)
-    if method_area == "xy":
-        if not (
-            mask.coord("projection_x_coordinate").has_bounds()
-            and mask.coord("projection_y_coordinate").has_bounds()
-        ):
-            mask.coord("projection_x_coordinate").guess_bounds()
-            mask.coord("projection_y_coordinate").guess_bounds()
-        area = np.outer(
-            np.diff(mask.coord("projection_x_coordinate").bounds, axis=1),
```

Comment:
There still seems to be an issue when using `method_area="xy"` for the area calculation. I tested it with the segmentation results of the OLR_Tracking_model notebook and got this error:

![Screenshot 2024-01-31 143227](https://github.com/tobac-project/tobac/assets/97912636/d72aa11f-5077-4ecf-992f-99bd0bcaa233)

Switching the lines in `area = np.outer(...)` fixes it on my end.

Reply:
Thanks for catching this, have fixed it now